lib/Dialect/TritonGPU/Transforms/RemoveLayoutConversions.cpp

```diff
@@ -1059,7 +1059,8 @@ void LayoutRematerialization::backwardRematerialization(
   // we don't handle conversions to DotOperandEncodingAttr
   // this is a heuristic to accommodate fused attention
   RankedTensorType targetType = convertOp.getType();
-  if (isa<DotOperandEncodingAttr>(targetType.getEncoding()))
+  auto dotEnc = dyn_cast<DotOperandEncodingAttr>(targetType.getEncoding());
```

Comment:
This part is not ready for a review. I want to see if any of the NVIDIA FA tests will fail, which will give me more context about the heuristic of not backward propagating conversions to dot layout. @ThomasRaoux (git blame led me to you :)) Is this related to mma -> dot conversion trick to keep result of the first dot in registers?

Reply:
Btw, for AMDGPU, IR is completely the same even if I completely remove this heuristics (not only for operand 1 as it is done in this patch).

Reply:
Ah, I think I understand what is the problem. 
If we allow dot layout to be backward propagated, in FA, for Q tensor, it will convert ops related to scaling of Q tensor from blocked to dot and as a byproduct, this will hoist conversion from blocked->dot (later blocked->shared, shared->dot), out of the loop. On AMD side, we do this anyway, but I think for NVIDIA this increases reg pressure too much so they avoid it. Propagating dot with opIdx == 1 which this patch needs isn't a problem (Q tensor has idx 0 in FA dot), however we probably want to fix this more generally. I will propose some solutions.

Reply:
Thanks for digging into it! It's fine to me right now to keep old behavior untouched and incrementally enable what we needed. Agreed we need to see if there is better way to solve it entirely.